frontend/device/generic/powerd.lua

```diff
@@ -133,18 +150,45 @@ function BasePowerD:setIntensity(intensity)
 end
 
 function BasePowerD:getCapacity()
-    local now_ts = os.time()
-    if now_ts - self.last_capacity_pull_time >= 60 then
-        self.battCapacity = self:getCapacityHW()
+    -- NOTE: UIManager *should* be loaded at this point.
+    --       If that doesn't hold, c.f., :stateChanged below.
+    local UIManager = require("ui/uimanager")
+    local now_ts = UIManager:getTime()
```

Comment:
Btw, I hack UIManager for my own tweaks, and call getCapacity() from in it. Your addition of require("uimanager") here to use UIManager:getTime() obviously causes me a crash:
```
./luajit: frontend/device/generic/powerd.lua:155: loop or previous error loading module 'ui/uimanager'
stack traceback:
    [C]: in function 'require'
    frontend/device/generic/powerd.lua:155: in function 'getCapacity'
    frontend/ui/uimanager.lua:2282: in function 'resetStateStats'
    frontend/ui/uimanager.lua:2128: in function '_initStandbyAndUnlock'
    frontend/ui/uimanager.lua:129: in function 'init'
    frontend/ui/uimanager.lua:2570: in main chunk
```
I'm going to hack me a solution, but I dunno, may be you'll feel too that requiring UIManager into a low level BasePowerD:getCapacity() just for getTime() is a bit awkward on a design PoV :)

Reply:
I clearly wouldn't have done it if the very function right below it didn't already do it ;p.